### DISCUSSION TITLE
Enable distinction between input sources: Automation, User, or External (e.g., original app)

### DISCUSSION THREAD

**Discussion by Felix-Kiefer** [home-assistant/frontend#23197] [Automations / scripts / scenes]
My Problem is currently, that I can't distinguish if a device or entity was turned on by an automation, a user input (like a button on the web panel that was manually triggered), or by an external app (for example: original app of the product, or through HomeKit). 

This limitation makes it difficult to implement logic that depends on knowing the exact origin of a state change. For example, automations might need to behave differently based on whether a device was turned on by a user, a specific automation, or an external app.

In my opinion, the perfect solution for that would be an additional "trigger" in the automation tab on the website, where I could select between event triggered by "user", "automation", "external change".

One main use case would be a smart self build thermostat controlled by Home Assistant:

1. User-initiated change: A user manually turns on the heating. A timer automation starts to turn it off after 15 minutes.

2. Automation-initiated change: A price-aware automation turns on the heating when electricity is cheap. No timer should start.

3. External change: The original app (outside of Home Assistant) turns on the heating. Home Assistant recognizes this and applies custom logic, such as notifying the user.

**Comment by ildar170975** [comment]
> I can't distinguish

Probably a "context" object is what you need to analyse.
